scripts/emulabconfig.py

```diff
@@ -1,27 +1,71 @@
+#!/usr/bin/env python
+"""
+This module defines an Emulab specific cluster hooks and exposes configuration information
+such as location of RAMCloud binaries and list of hosts. EMULAB_HOST environment must be 
+exported to point to node-0 in the Cloudlab experiment where the caller has passwordless
+ssh access. Check https://www.cloudlab.us/ssh-keys.php if you didn't export keys already
+"""
+
 import subprocess
 import sys
 import os
 import re
 import socket
+import xml.etree.ElementTree as ET
+
+__all__ = ['getHosts', 'local_scripts_path', 'top_path', 'obj_path',
+           'default_disk1', 'default_disk2', 'EmulabClusterHooks', 'log'] 
+
+hostname = socket.gethostname()
 
-hosts = [
-            ('pc813.emulab.net', '155.98.36.113', 0),
-            ('pc806.emulab.net', '155.98.36.106', 1),
-            ('pc807.emulab.net', '155.98.36.107', 2),
-            ('pc805.emulab.net', '155.98.36.105', 3)
-        ]
+def log(msg):
+    print '[%s] %s' % (hostname, msg)
+
+
+# If run locally, connects to EMULAB_HOST and gets the manifest from there to
+# populate host list, since this is invoked to compile RAMCloud (rawmetrics.py)
+# the default is to see if you can get the manifest locally
+def getHosts():
+    nodeId = 0
+    serverList = []
+    try:
+        log("trying to get manifest locally")
+        out = captureSh("/usr/bin/geni-get manifest",shell=True, stderr=subprocess.STDOUT)
+    except:
+        log("trying EMULAB_HOST to get manifest")
+        if 'EMULAB_HOST' not in os.environ:
+            log("'EMULAB_HOST' not exported")
+            sys.exit(1)
+        out = subprocess.check_output("ssh %s /usr/bin/geni-get manifest" % os.environ['EMULAB_HOST'],
+                                       shell=True, stderr=subprocess.STDOUT)
+
+    root = ET.fromstring(out)
+    for child in root.getchildren():
+        if child.tag.endswith('node'):
+            for host in child.getchildren():
+                if host.tag.endswith('host'):
+                    serverList.append((host.attrib['name'], host.attrib['ipv4'], nodeId))
+                    nodeId += 1
+    return serverList
 
 def ssh(server, cmd, checked=True):
+    """ Runs command on a remote machine over ssh.""" 
     if checked:
         return subprocess.check_call('ssh %s "%s"' % (server, cmd),
                                      shell=True, stdout=sys.stdout)
     else:
         return subprocess.call('ssh %s "%s"' % (server, cmd),
                                shell=True, stdout=sys.stdout)
 
-hostname = socket.gethostname()
-def log(msg):
-    print '[%s] %s' % (hostname, msg)
+
+def pdsh(cmd, checked=True):
+    """ Runs command on remote hosts using pdsh on remote hosts"""
+    if checked:
+        return subprocess.check_call('pdsh -w^/tmp/emulab-hosts "%s"' % cmd,
```

Comment:
You may want to write this to a file somewhere in the cwd and add the file name to .gitignore. Otherwise, if multiple people run the script on the same host (royal), then they'll collide.